### PR TITLE
refactor(js): start preparing JS codebase for Flow's types-first mode

### PR DESCRIPTION
**Pull request by mcous**
## overview

Flow has a large-scale architectural change that just came out of experimental mode that promises significant type-check speed-up (including orders of magnitude speed-ups in some cases). This change is called "types first" and you can read about it here:

- https://medium.com/flow-type/types-first-a-scalable-new-architecture-for-flow-3d8c7ba1d4eb
- https://flow.org/en/docs/lang/types-first/

In order to enable types first mode in our codebase, we have to explicitly annotate the type of anything that a given module exports; the speedup comes when flow no longer has to try to infer types across module boundaries.

This PR does not (and cannot) enable types-first mode. It is one in a series of at least 3 or 4 PRs to gradually get this mode switched on

## changelog

This PR prepares our library code for types first by ensuring that all exports are annotated. It does **not touch `app`, `labware-library`, or `protocol-designer`**. Additionally, it **should not touch anything other than type annotations**.

- Upgrade Flow to v0.125 for non-experimental types-first support (no new type-check failures)
- Enable `well_formed_exports` option to complain if any exports are not annotated
    - Enable it for everything except `app`, `labware-library`, or `protocol-designer`
- I lied about not changing source code! I removed the following unused components from the CL
    - `ListAlert`
    - `IntervalWrapper`

## review requests

- [ ] Dev tasks pass
- [ ] **No source code changes made**; diff should be Flow-only changes
    - Aside from unused component removals

## risk assessment

Very low if the above two points are true because type annotations do not affect runtime

### CODE REVIEW

File: shared-data/pipette/fixtures/name/index.js [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
 
-export const fixtureP10Single = pipetteNameSpecFixtures['p10_single']
-export const fixtureP10Multi = pipetteNameSpecFixtures['p10_multi']
-export const fixtureP300Single = pipetteNameSpecFixtures['p300_single']
-export const fixtureP300Multi = pipetteNameSpecFixtures['p300_multi']
-export const fixtureP1000Single = pipetteNameSpecFixtures['p1000_single']
-export const fixtureP1000Multi = pipetteNameSpecFixtures['p1000_multi']
+import type { PipetteNameSpecs } from '../../../js/types'
+
+export const fixtureP10Single: $Shape<PipetteNameSpecs> =
```
Comment (IanLondon) [inline shared-data/pipette/fixtures/name/index.js:6]: why `$Shape` and not just `PipetteNameSpecs`?

**Review by IanLondon — APPROVED** [review]
all looks good, thanks!

File: shared-data/pipette/fixtures/name/index.js (hunk 1 above)
Comment (mcous) [reply shared-data/pipette/fixtures/name/index.js:6]: I compared the JSON file with the actual spec and found that the JSON file was missing fields compared to the actual type. Calling it a slightly unsafe `$Shape` felt like the most accurate thing to do, but open to suggestions

Comment (IanLondon) [reply shared-data/pipette/fixtures/name/index.js:6]: oh weird, I didn't realize it was missing fields. I thought there were jest tests making sure they conform to schema, but maybe pipette fixtures don't have those

**Pull request merged**